### ISSUE TITLE
Navigation scripts are not well-encapsulated

### ISSUE THREAD

**Issue by jmuzina** [canonical/vanilla-framework#5180]
**Describe the bug**

The JS scripts used by the navigation pattern do not work properly when there are multiple navigation components on the screen. This introduces issues for #5165 , where we would like the nav component to be fully interactive in combined form, just as it is in singular form. 

**To Reproduce**

1. Add multiple navigation components to the same page. Import the same navigation scripts in each component separately. A good way to do this is to checkout the [combined navigation PR](https://github.com/canonical/vanilla-framework/pull/5165) and then revert the [commit that disabled its interactivity](https://github.com/canonical/vanilla-framework/pull/5165/commits/86a5aab39e073b6c14a85b8441f7ca51a9ad2ca4). 
2. Observe console error: 
```
Uncaught SyntaxError: redeclaration of const initNavigationSliding
```

**Expected behavior**

Multiple imports of the navigation scripts should not cause errors. The component's script should be encapsulated such that there can be multiple indepedent navigation components on the same DOM.

**Comment by bartaz** [comment]
@jmuzina is this still relevant? I think we disabled scripting in combined examples, right?

**Comment by jmuzina** [comment]
> @jmuzina is this still relevant? I think we disabled scripting in combined examples, right?

@bartaz 

Yes, We have disabled including the script block in combined examples. So, this is technically solved.
